cpp/src/arrow/compute/exec/hash_join_benchmark.cc

```diff
@@ -0,0 +1,417 @@
+      build_metadata["min"] = std::to_string(min_build_value);
+      build_metadata["max"] = std::to_string(max_build_value);
+
+      std::unordered_map<std::string, std::string> probe_metadata;
+      probe_metadata["null_probability"] = std::to_string(settings.null_percentage);
+      probe_metadata["min"] = std::to_string(min_probe_value);
+      probe_metadata["max"] = std::to_string(max_probe_value);
+
+      auto l_field =
+          field(l_name, settings.key_types[i], key_value_metadata(probe_metadata));
+      auto r_field =
+          field(r_name, settings.key_types[i], key_value_metadata(build_metadata));
+
+      DCHECK_OK(l_schema_builder.AddField(l_field));
+      DCHECK_OK(r_schema_builder.AddField(r_field));
+
+      left_keys.push_back(FieldRef(l_name));
+      right_keys.push_back(FieldRef(r_name));
+    }
+
+    for (size_t i = 0; i < settings.build_payload_types.size(); i++) {
+      std::string name = "lp" + std::to_string(i);
+      DCHECK_OK(l_schema_builder.AddField(field(name, settings.probe_payload_types[i])));
+    }
+
+    for (size_t i = 0; i < settings.build_payload_types.size(); i++) {
+      std::string name = "rp" + std::to_string(i);
+      DCHECK_OK(r_schema_builder.AddField(field(name, settings.build_payload_types[i])));
+    }
+
+    auto l_schema = *l_schema_builder.Finish();
+    auto r_schema = *r_schema_builder.Finish();
+
+    l_batches_ =
+        MakeRandomBatches(l_schema, settings.num_probe_batches, settings.batch_size);
+    r_batches_ =
+        MakeRandomBatches(r_schema, settings.num_build_batches, settings.batch_size);
+
+    stats_.num_probe_rows = settings.num_probe_batches * settings.batch_size;
+
+    ctx_ = arrow::internal::make_unique<ExecContext>(
+        default_memory_pool(),
+        is_parallel ? arrow::internal::GetCpuThreadPool() : nullptr);
+
+    schema_mgr_ = arrow::internal::make_unique<HashJoinSchema>();
+    Expression filter = literal(true);
+    DCHECK_OK(schema_mgr_->Init(settings.join_type, *l_batches_.schema, left_keys,
+                                *r_batches_.schema, right_keys, filter, "l_", "r_"));
+
+    join_ = *HashJoinImpl::MakeBasic();
+
+    omp_set_num_threads(settings.num_threads);
+    auto schedule_callback = [](std::function<Status(size_t)> func) -> Status {
+#pragma omp task
+      { DCHECK_OK(func(omp_get_thread_num())); }
+      return Status::OK();
+    };
+
+    DCHECK_OK(join_->Init(
+        ctx_.get(), settings.join_type, !is_parallel /* use_sync_execution*/,
+        settings.num_threads, schema_mgr_.get(), {JoinKeyCmp::EQ}, std::move(filter),
+        [](ExecBatch) {}, [](int64_t x) {}, schedule_callback));
+  }
+
+  void RunJoin() {
+#pragma omp parallel
+    {
+      int tid = omp_get_thread_num();
+#pragma omp for nowait
+      for (auto it = r_batches_.batches.begin(); it != r_batches_.batches.end(); ++it)
+        DCHECK_OK(join_->InputReceived(tid, 1 /* side */, *it));
```

Comment:
```suggestion
        DCHECK_OK(join_->InputReceived(tid, /*side=*/1, *it));
```
For consistency `foo(/*param_name=*/0)` is more common in the code base than `foo(0 /* param_name */)`